dendrite-config.yaml

```diff
@@ -0,0 +1,73 @@
+# The config file version format
+version: v0
```

Comment:
I've made it an int and chucked some more comments in there. 9139153
(Saying something is "nicer" is not necessarily helpful here. I'd be interested in what you think you can do with an int that you can't do with a string. In this case I'm assuming it's because you can distinguish between it being too new and being too old when erroring at startup.)